nanshe_ipython.ipynb

```diff
@@ -1074,14 +1078,20 @@
         "\n",
         "mskctr_srcs.callback = CustomJS(\n",
         "    args=dict(\n",
+        "        all_tr_dtype_srcs=all_tr_dtype_srcs,\n",
         "        all_tr_shape_srcs=all_tr_shape_srcs,\n",
         "        all_tr_srcs=all_tr_srcs,\n",
         "        tr_srcs=tr_srcs\n",
         "    ), code=\"\"\"\n",
         "    var range = function(n){ return Array.from(Array(n).keys()); };\n",
         "\n",
         "    var inds = cb_obj.get('selected')['1d'].indices;\n",
-        "    var traces = all_tr_srcs.get('data')['traces'];\n",
+        "    var traces_dtype = all_tr_dtype_srcs.get('data')['traces_dtype'].constructor;\n",
+        "    try {\n",
+        "        var traces = new traces_dtype(window.pako.inflate(all_tr_srcs.get('data')['traces']).buffer);\n",
```

Comment:
First, decompress the data with pako. The result will also be a `Uint8Array` just as the input is. Then take the underlying `buffer` of the `Uint8Array` and create a view onto it using a `TypedArray` matching the original type of the data. As we create a view onto the underlying data, we avoid a potentially expensive copy in this step.